discovery-infra/tests/test_validations.py

```diff
@@ -0,0 +1,76 @@
+        self.assert_cluster_validation(orig_cluster, "network", "cluster-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "service-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "no-cidrs-overlapping", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "network-prefix-valid", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "ntp-server-configured", "success")
+
+        # Check base DNS domain
+        api_client.update_cluster(cluster_id, {"base_dns_domain": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["failure"])
+        api_client.update_cluster(cluster_id, {"base_dns_domain": orig_cluster.base_dns_domain})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["success"])
+
+        # Check pull secret
+        api_client.update_cluster(cluster_id, {"pull_secret": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["failure"])
+        api_client.update_cluster(cluster_id, {"pull_secret": env_variables['pull_secret']})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["success"])
+
+    def test_host_insufficient(self, api_client, cluster, nodes):
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+
+        # Modify vCPU count on the first node to be insufficient
+        vcpu_node = nodes.nodes[0]
+        orig_vcpu_value = vcpu_node.get_cpu_cores()
+        vcpu_node.set_cpu_cores(1)
+        vcpu_host_id = vcpu_node.get_host_id()
+
+        # Modify RAM amount on the second node to be insufficient
+        ram_node = nodes.nodes[1]
+        ram_node.set_ram_kib(2097152)  # 2GB
+        ram_host_id = ram_node.get_host_id()
+
+        nodes.start_all()
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client, additional_statuses=[consts.NodesStatus.INSUFFICIENT])
+        self.wait_for_host_validation(api_client, cluster_id, vcpu_host_id, "hardware", "has-min-cpu-cores", ["failure"])
+        self.wait_for_host_validation(api_client, cluster_id, ram_host_id, "hardware", "has-min-memory", ["failure"])
+
+        # Return to original settings and verify that validations pass
+        vcpu_node.shutdown()
+        vcpu_node.set_cpu_cores(orig_vcpu_value)
+        vcpu_node.start()
+
+        ram_node.shutdown()
+        ram_node.set_ram_kib(int(env_variables['master_memory']) * 1024)
+        ram_node.start()
+
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client, additional_statuses=[consts.NodesStatus.INSUFFICIENT])
+        self.wait_for_host_validation(api_client, cluster_id, vcpu_host_id, "hardware", "has-min-cpu-cores", ["success"])
```

Comment:
Maybe the strings part will benefit from keyword arguments, to explain what ``"hardware"``, ``"has-min-cpu-cores"`` and ``["success"]`` are exactly

Reply:
I think it's clear, given the method deals with validations?